srk/src/polynomialCone.ml

```diff
@@ -126,47 +118,20 @@ let add_polys_to_cone pc zero_polys nonneg_polys =
 let trivial = (Rewrite.mk_rewrite Monomial.degrevlex [QQXs.one], [])
 
 let cone_of_polyhedron poly ctx =
-  Log.errorf "cone of polyhedron";
-  let dim = PVCTX.num_dimensions ctx in
-  (* let dim = MonomialMap.cardinal pvutil.mono_map in *)
-  let cone_generators = BatList.concat_map (fun (typ, vec_generator) ->
-      match typ with
-      |  `Ray ->
-        logf "ray generator %a" V.pp vec_generator;
-        let p =BatEnum.fold (fun p (coeff, index) ->
-            QQXs.add_term coeff (PVCTX.monomial_of index ctx) p)
-            (QQXs.zero)
-            (Linear.QQVector.enum vec_generator)
-        in [p]
-      (* deal with lines as a pair of ray generation  *)
-      | `Line -> logf "line generator %a" V.pp vec_generator;
-        let p1 = BatEnum.fold (fun p (coeff, index) ->
-            QQXs.add_term coeff (PVCTX.monomial_of index ctx) p)
-            (QQXs.zero)
-            (Linear.QQVector.enum vec_generator)
-        in
-        let p2 = BatEnum.fold (fun p (coeff, index) ->
-            QQXs.add_term (QQ.negate coeff) (PVCTX.monomial_of index ctx) p)
-            (QQXs.zero)
-            (Linear.QQVector.enum vec_generator)
-        in
-        [p1; p2]
-      (* should always have one vertex at 0 *)
-      | `Vertex -> [QQXs.zero]
-    )
-      (BatList.of_enum (Polyhedron.enum_generators dim poly))
-  in
+  let zero_polys, geq_zero_polys = extract_polynomial_constraints ctx poly in
+  let zero_polys = BatList.concat_map (fun p -> [QQXs.negate p; p]) zero_polys in
+  let cone_generators = zero_polys @ geq_zero_polys in
   BatList.filter (fun p -> not (QQXs.equal p (QQXs.zero))) cone_generators
 
 let project_cone cone_generators f =
   Log.errorf "computing projection of cone part";
   let cone_generators = QQXs.one :: cone_generators in
-  let ctx = PVCTX.mk_context ~fix_const_dim:false Monomial.degrevlex cone_generators in
+  let ctx = PVCTX.mk_context Monomial.degrevlex cone_generators in
   let dim = PVCTX.num_dimensions ctx in
   Log.errorf "num dimension is: %d" dim;
   let polyhedron_rep_of_cone = polyhedron_of_cone cone_generators ctx in
   Log.errorf "polyhedron of cone before proj: %a" (Polyhedron.pp pp_dim) polyhedron_rep_of_cone;
-  let elim_dims = (0 -- (dim - 1)) |> BatEnum.filter
+  let elim_dims = (1 -- (dim - 1)) |> BatEnum.filter
```

Comment:
Why starting at 1 instead of 0?

Reply:
Now monomial 1 is mapped to dimension -1 in polynomialUtil. And other monomials start from dimension 1 (rather than 0). So I think dimension 0 is not mapped to any monomial currently (indeed it will raise the exception if we start from 0)